client/rest-high-level/src/test/java/org/elasticsearch/client/asyncsearch/AsyncSearchIT.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.asyncsearch;
+
+import org.elasticsearch.client.ESRestHighLevelClientTestCase;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public class AsyncSearchIT extends ESRestHighLevelClientTestCase {
+
+    public void testSubmitAsyncSearchRequest() throws IOException {
+        String index = "test-index";
+        createIndex(index, Settings.EMPTY);
+
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder().query(QueryBuilders.matchAllQuery());
+        SubmitAsyncSearchRequest request = new SubmitAsyncSearchRequest(sourceBuilder, index);
+        // 2 sec should be enough to make sure we always complete right away
+        request.setWaitForCompletion(new TimeValue(2, TimeUnit.SECONDS));
```

Comment:
this should work but it's also risky. What other options we have? 
1) increase the timeout even more to really make sure it's enough, though what is enough
2) accept that we may get a response while the search is still running, in which case is_partial and is_final will have a different value? Maybe don't set cleanOnCompletion and assert that when the id is returned the search is running, and when the id is not returned, both flags are false.

I am not sure either way, maybe  I would go for option 1.

Reply:
We can also have a loop here when get is implemented that submits the initial request and use get until `isRunning` is false on the response ?

Reply:
I'll go with 1.) then, staying on the save side. I will keep `cleanOnCompletion` though, currently thats the only way I can see to actually get an Id and I'd like to tests its deserialization etc... here if possible (will also need ids in a follow-up PR around get-API)

Reply:
sorry, missed the previous comment. Loop sounds okay as well, getting a response thats not running any more isn't actually the problem, the oposite is tricky (getting a response with an Id, as I mentioned earlier). I don't need that here so will remove it but will need it later for get/delete API tests.

Reply:
Just also realize I haven't implemented get yet in this PR, we could change this later to a loop with get if you agree? 
I don't see the benefits over a long-enough wait time though, quite the oposite. I never got this call to _not_ finish in the first call, neither locally nor on CI which is even faster.

Reply:
I like the loop idea if it does not make things too complicated, it sounds doable.

Reply:
it's fine to make this a loop later ;)